### ISSUE TITLE
Pyright 1.1.344 reports that a `strawberry.scalar(...)` variable cannot be used for return type annotation

### ISSUE THREAD

**Issue by kkom** [strawberry-graphql/strawberry#3312]
## Describe the Bug

See this toy example schema with the newly detected type error in comments:

```python
Slug = strawberry.scalar(
    str,
    serialize=lambda x: x,
    parse_value=lambda x: x,
)

@strawberry.type
class Organisation:
    @strawberry.field
    def slug(self) -> Slug:  # error: Variable not allowed in type expression (reportGeneralTypeIssues)
        ...
```

Pyright complains that a variable is not allowed in type expression, which seems like a legit complaint to me.

The custom scalar type use case can be probably achieved more elegantly / correctly w.r.t. static types using `typing.Annotated`?

## System Information

`strawberry`: `0.217.0`
`pyright`: `1.1.344` (not detected in `1.1.343`)

**Comment by patrick91** [comment]
> The custom scalar type use case can be probably achieved more elegantly / correctly w.r.t. static types using typing.Annotated?

yup, I think we need to do this, it's just annoying to update all the places (we did a similar thing for unions: https://github.com/strawberry-graphql/strawberry/pull/2840)
